app/src/main/java/org/simple/clinic/editpatient/PatientEditScreenController.kt

```diff
@@ -406,64 +379,9 @@ class PatientEditScreenController @Inject constructor(
     return confirmDiscardChanges.mergeWith(closeScreenWithoutConfirmation)
   }
 
-  private fun hasEdits(
-      savedPatient: Patient,
-      savedPatientPhoneNumber: PatientPhoneNumber?,
-      savedPatientAddress: PatientAddress,
-      updatedEntry: OngoingEditPatientEntry,
-      savedEntry: OngoingEditPatientEntry
-  ): Boolean {
-    return hasPatientChanged(updatedEntry, savedPatient)
-        || hasPhoneNumberChanged(savedPatientPhoneNumber, updatedEntry)
-        || hasColonyOrVillageChanged(savedPatientAddress, updatedEntry)
-        || hasDistrictOrStateChanged(updatedEntry, savedPatientAddress)
-        || hasAgeOrDateOfBirthChanged(updatedEntry, savedPatient)
-  }
-
-  private fun hasPatientChanged(entry: OngoingEditPatientEntry, savedPatient: Patient) =
-      entry.name != savedPatient.fullName || entry.gender != savedPatient.gender
-
-  private fun hasPhoneNumberChanged(savedPatientPhoneNumber: PatientPhoneNumber?, entry: OngoingEditPatientEntry): Boolean {
-    return when (savedPatientPhoneNumber) {
-      null -> entry.phoneNumber.isNotBlank()
-      else -> savedPatientPhoneNumber.number != entry.phoneNumber
-    }
-  }
-
-  private fun hasColonyOrVillageChanged(savedPatientAddress: PatientAddress, entry: OngoingEditPatientEntry): Boolean {
-    return when (savedPatientAddress.colonyOrVillage) {
-      null -> entry.colonyOrVillage.isNotBlank()
-      else -> savedPatientAddress.colonyOrVillage != entry.colonyOrVillage
-    }
-  }
-
-  private fun hasDistrictOrStateChanged(entry: OngoingEditPatientEntry, savedPatientAddress: PatientAddress) =
-      entry.district != savedPatientAddress.district || entry.state != savedPatientAddress.state
-
-  private fun hasAgeOrDateOfBirthChanged(entry: OngoingEditPatientEntry, savedPatient: Patient): Boolean {
-    return when (entry.ageOrDateOfBirth) {
-      is EntryWithAge -> hasAgeChanged(savedPatient, entry.ageOrDateOfBirth)
-      is EntryWithDateOfBirth -> hasDateOfBirthChanged(savedPatient, entry.ageOrDateOfBirth)
-    }
-  }
-
-  private fun hasAgeChanged(savedPatient: Patient, ageOrDateOfBirth: EntryWithAge): Boolean {
-    val savedAgeAsString = savedPatient.age?.value?.toString()
-    return if (savedAgeAsString == null && ageOrDateOfBirth.age.isBlank()) {
-      false
-    } else {
-      ageOrDateOfBirth.age != savedAgeAsString
-    }
-  }
-
-  private fun hasDateOfBirthChanged(savedPatient: Patient, ageOrDateOfBirth: EntryWithDateOfBirth): Boolean {
-    val savedDateOfBirthAsString = savedPatient.dateOfBirth?.format(dateOfBirthFormatter)
-
-    return if (savedDateOfBirthAsString == null && ageOrDateOfBirth.dateOfBirth.isBlank()) {
-      false
-
-    } else {
-      ageOrDateOfBirth.dateOfBirth != savedDateOfBirthAsString
-    }
+  private inline fun <reified T : Any, R> Observable<UiEvent>.mapType(
```

Comment:
Nice!

This would be a useful function to have in other places. Can we move this to `RxFoo.kt`?